test/gui/shared/steps/steps.py

```diff
@@ -529,8 +527,20 @@ def step(context, resource, role):
         radioObjectName = names.oCC_ShareLinkWidget_radio_uploadOnly_QRadioButton
     else:
         raise Exception("No such role found for resource")
-           
+
     openPublicLinkDialog(context, resource)
     clickButton(waitForObject(radioObjectName))
-    clickButton(waitForObject(names.oCC_ShareLinkWidget_createShareButton_QPushButton))       
+    clickButton(waitForObject(names.oCC_ShareLinkWidget_createShareButton_QPushButton))  
+    
+@When('the user creates a new public link for folder "|any|" using the client-UI with these details:')
+def step(context, resource):
+    for row in context.table:
+        if row[0] == 'role':       
+            createPublicShare(context, resource, row[1])
```

Comment:
that would only create the publiLink if the "role" is given and and if multiple "role" lines are given it would create multiple public links. no other settings are are taken in account

Reply:
This step actual is used just to create a public link based on role. For multiple role line I can add a break so that it works for only one role.